lib/ansible/cli/adhoc.py

```diff
@@ -64,10 +64,18 @@ def post_process_args(self, options):
 
         return options
 
-    def _play_ds(self, pattern, async_val, poll):
-        check_raw = context.CLIARGS['module_name'] in C.MODULE_REQUIRE_ARGS
-
-        mytask = {'action': {'module': context.CLIARGS['module_name'], 'args': parse_kv(context.CLIARGS['module_args'], check_raw=check_raw)},
+    def _play_ds(self, pattern, async_val, poll, loader):
```

Comment:
do not use loader, there are from_yaml/json functions you should use instead, loader has many other implications i would not want to apply to this case.

Reply:
ack will use basic json.load